dynamic-consumer/src/main/scala/nl/vroste/zio/kinesis/client/dynamicconsumer/SchedulerConfig.scala

```diff
@@ -28,29 +29,38 @@ case class SchedulerConfig(
   initialPositionInStreamExtended: InitialPositionInStreamExtended =
     InitialPositionInStreamExtended.newInitialPosition(InitialPositionInStream.LATEST),
   private val kinesisClient: KinesisAsyncClient,
-  private val streamName: String
+  private val streamIdentifier: StreamIdentifier
 ) {
   def withInitialPosition(pos: InitialPositionInStreamExtended): SchedulerConfig =
     copy(
       leaseManagement = leaseManagement.initialPositionInStream(pos),
       initialPositionInStreamExtended = pos
     )
 
-  def withEnhancedFanOut: SchedulerConfig =
+  def withEnhancedFanOut: SchedulerConfig = {
+    val config           = new FanOutConfig(kinesisClient)
+      .applicationName(retrieval.applicationName())
+    val configWithStream = streamIdentifier match {
+      case StreamIdentifier.StreamIdentifierByName(streamName) => config.streamName(streamName)
+      case StreamIdentifier.StreamIdentifierByArn(_)           => ??? // TODO
```

Comment:
This API doesn't seem to support ARNs - I'm not sure what the best solution is.

Reply:
Perhaps for now we'll have to require a StreamIdentifierByName as parameter to `shardedStream`